apps/docs/content/guides/auth/server-side/nextjs.mdx

```diff
@@ -415,6 +415,26 @@ Create a Route Handler for `auth/confirm`. When a user clicks their confirmation
 
 Since this is a Router Handler, use the Supabase client from `@/utils/supabase/server.ts`.
 
+<Admonition type="caution">
+  Attention: in certain specific cases, such as with Microsoft's email provider, it is possible that links in emails are pre-opened (e.g., [Safe Links in Microsoft Defender for Office 365](https://learn.microsoft.com/en-us/defender-office-365/safe-links-about?view=o365-worldwide)). This can result in the confirmation link being opened before the user clicks on it, consuming the confirmation token and rendering it expired. To perform this action, email services send an HTTP HEAD request to check the validity of the link.
+
+  To avoid this issue, a double confirmation system can be implemented. For example, by directing the user to a confirmation page where they need to click a button to confirm their email, as explained [here](/docs/guides/auth/auth-email-templates#limitations).
+
+  Another solution is to "block" HTTP HEAD requests (thus preventing the token from being consumed) by adding this code at the beginning of your route:
+
+  ```ts app/auth/confirm/route.ts
+  export async function GET(request: NextRequest) {
+    if (request.method !== 'GET') {
+        return new NextResponse(
+          'Method Not Allowed', 
+          { status: 405 }
+        );
+    }
+    // Rest of the code
+  }
+  ```
+</Admonition>
```

Comment:
@thibault-chausson do you have a reproducible setup that proves that this works?

Reply:
Hello @kangmingtay,

I'm sorry, I don't have an example code to reproduce this behavior. Additionally, I don't have much time at the moment to code an example. If you have code with authentication in App Router for NextJS, I could take a look at it.

However, the proposed code with (lignes 11 - 14):

<img width="1127" alt="Screenshot 2024-07-21 at 10 57 32" src="https://github.com/user-attachments/assets/854d1802-84de-447e-9f89-ef55fb7349fe">


Is inspired by the code for Page Router applications of NextJS as proposed in the Supabase documentation: https://supabase.com/docs/guides/auth/server-side/nextjs?queryGroups=router&router=pages

I remain available if needed.

Have a good day,